test/cluster/catchup_test.go

```diff
@@ -0,0 +1,383 @@
+		t.Logf("key-%+v", key)
+		require.NoError(t, err)
+		require.NotNil(t, rcpt)
+		require.True(t, txID != "")
+		require.True(t, len(rcpt.GetHeader().GetValidationInfo()) > 0)
+		require.Equal(t, types.Flag_VALID, rcpt.Header.ValidationInfo[rcpt.TxIndex].Flag)
+		txsCount++
+		t.Logf("tx submitted: %s, %+v", txID, rcpt)
+	}
+	var dataEnv *types.GetDataResponseEnvelope
+	for _, key := range keys {
+		require.Eventually(t, func() bool {
+			dataEnv, err = c.Servers[leaderRound1].QueryData(t, worldstate.DefaultDBName, strconv.Itoa(key))
+			return dataEnv != nil && dataEnv.GetResponse().GetValue() != nil && err == nil
+		}, 30*time.Second, 100*time.Millisecond)
+		dataResp := dataEnv.GetResponse().GetValue()
+		require.Equal(t, dataResp, data)
+	}
+	require.Eventually(t, func() bool {
+		return c.AgreedHeight(t, uint64(txsCount+1), 0, 1, 2)
+	}, 30*time.Second, 100*time.Millisecond)
+
+	follower1 := (leaderRound1 + 1) % 3
+	follower2 := (leaderRound1 + 2) % 3
+
+	require.NoError(t, c.ShutdownServer(c.Servers[leaderRound1]))
+
+	//find new leader and make sure there are only 2 active nodes
+	leaderRound2 := -1
+	require.Eventually(t, func() bool {
+		leaderRound2 = c.AgreedLeader(t, follower1, follower2)
+		return leaderRound2 >= 0
+	}, 30*time.Second, 100*time.Millisecond)
+
+	t.Logf("Stopped node %d, new leader index is: %d; 2-node quorum", leaderRound1, leaderRound2)
+
+	follower1Round2 := follower1
+	follower2Round2 := leaderRound1
+	if leaderRound2 == follower1 {
+		follower1Round2 = follower2
+	}
+
+	//servers alive: leaderRound2, follower1Round2
+	require.Eventually(t, func() bool {
+		clusterStatusResEnv, err := c.Servers[leaderRound2].QueryClusterStatus(t)
+		return err == nil && clusterStatusResEnv != nil && len(clusterStatusResEnv.GetResponse().GetActive()) == 2
+	}, 30*time.Second, 100*time.Millisecond)
+
+	victimServer, activeServer := follower1Round2, leaderRound2
+	if victimIsLeader {
+		victimServer, activeServer = leaderRound2, follower1Round2
+	}
+	require.NoError(t, c.ShutdownServer(c.Servers[victimServer]))
+
+	//only one server is active now => no majority to pick leader
+	require.Eventually(t, func() bool {
+		_, _, err = c.Servers[activeServer].WriteDataTx(t, worldstate.DefaultDBName, "bob", data)
+		return err != nil && err.Error() == "failed to submit transaction, server returned: status: 503 Service Unavailable, message: Cluster leader unavailable"
+	}, 60*time.Second, 100*time.Millisecond)
+
+	//restart one server => 2 servers are active: follower1Round2, leaderRound2
+	require.NoError(t, c.StartServer(c.Servers[victimServer]))
+
+	//find the new leader
+	leaderRound3 := -1
+	require.Eventually(t, func() bool {
+		leaderRound3 = c.AgreedLeader(t, leaderRound2, follower1Round2)
+		return leaderRound3 >= 0
+	}, 30*time.Second, 100*time.Millisecond)
+
+	t.Logf("Started node %d, leader index is: %d; 2-node quorum", follower1Round2, leaderRound3)
```

Comment:
the started server is `victimServer`, not `follower1Round2`

Reply:
Done